File: src/schema/application/ip_infos.ts
import { pgTable, text, index } from 'drizzle-orm/pg-core';
import {
  getConfigDefaults,
  system_fields,
} from '@dna-platform/crdt-lww-postgres/build/schema/system';
import { primaryKey } from 'drizzle-orm/pg-core';
const config = (table) => ({
  pk: primaryKey({ columns: [table.id] }),
  ...getConfigDefaults.defaultIndexes('ip_infos', table),
  ip_infos_ip_idx: index('ip_infos_ip_idx').on(table.ip),
  ip_infos_country_idx: index('ip_infos_country_idx').on(table.country),
});
export const table = pgTable(
  'ip_infos',
  {
    ...system_fields,
    id: text('id'), // Primary key ---
    // timestamp: timestamp('timestamp', { withTimezone: true }), // NOT NULL timestamp
    ip: text('ip').unique(),
    country: text('country'),
    asn: text('asn'),
    org: text('org'),
    continent_code: text('continent_code'),
    city: text('city'),
    region: text('region'),
    postal: text('postal'),
    timezone: text('timezone'),
  },
  config,
);

File: bin/store/src/database/schema/tables/ip_infos.rs
use crate::define_table_schema;
use crate::generated::schema::generator::diesel_schema_definition::{
    types::*, DieselTableDefinition,
};
use crate::{system_fields, system_foreign_keys, system_indexes};

/**
 * Auto-converted from TypeScript schema source.
 */
pub struct IpInfosTable;

define_table_schema! {
    hypertable: false,
    fields: {
        // System fields - common across all tables ( REQUIRED )
        system_fields!(),

        ip: nullable(text()),
        country: nullable(text()),
        asn: nullable(text()),
        org: nullable(text()),
        continent_code: nullable(text()),
        city: nullable(text()),
        region: nullable(text()),
        postal: nullable(text()),
        timezone: nullable(text()),
    },
    indexes: {
        // System field indexes ( REQUIRED )
        system_indexes!("ip_infos"),
        // Custom table-specific indexes
        idx_ip_infos_ip: {
            columns: ["ip"],
            unique: false,
            type: "btree"
        },
        idx_ip_infos_country: {
            columns: ["country"],
            unique: false,
            type: "btree"
        },
    },
    foreign_keys: {
        // System field foreign keys ( REQUIRED )
        system_foreign_keys!("ip_infos"),
    }
}
